stable/yugabyte/templates/certificates.yaml

```diff
@@ -0,0 +1,122 @@
+{{- $root := . -}}
+---
+{{- if $root.Values.tls.certManager.enabled }}
+{{- if not $root.Values.tls.useClusterIssuer }}
+---
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name:  {{ (printf "%s-bootstrap" (include "yugabyte.tls_cm_issuer" $root)) | quote }}
+  namespace: "{{ $root.Release.Namespace }}"
+spec:
+  selfSigned: {}
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: yugabyte-ca
+  namespace: "{{ $root.Release.Namespace }}"
+spec:
+  isCA: true
+  privateKey:
+    algorithm: ECDSA
+    size: 521
+  commonName: Yugabyte Selfsigned CA
+  secretName: yugabyte-ca
+  issuerRef:
+    name: {{ (printf "%s-bootstrap" (include "yugabyte.tls_cm_issuer" $root)) | quote }}
+    kind: Issuer
+---
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: {{ include "yugabyte.tls_cm_issuer" $root | quote }}
+  namespace: "{{ $root.Release.Namespace }}"
+spec:
+  ca:
+    secretName: yugabyte-ca
+---
+{{- end }}
+
+
+{{- range .Values.Services }}
+{{- $service := . -}}
+{{- $appLabelArgs := dict "label" .label "root" $root -}}
+{{- $serviceValues := (dict "Service" $service "Values" $root.Values "Chart" $root.Chart "Release" $root.Release) -}}
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: {{ $root.Values.oldNamingStyle | ternary (printf "%s-yugabyte-tls-cert" $service.label) (printf "%s-%s-tls-cert" (include "yugabyte.fullname" $root) $service.label) }}
+  namespace: "{{ $root.Release.Namespace }}"
+spec:
+  secretTemplate:
+    labels:
+      {{- include "yugabyte.applabel" ($appLabelArgs) | indent 6 }}
+      {{- include "yugabyte.labels" $root | indent 6 }}
+  issuerRef:
+    name:  {{ include "yugabyte.tls_cm_issuer" $root | quote }}
+    {{- if $root.Values.tls.certManager.useClusterIssuer }}
+    kind: ClusterIssuer
+    {{- else }}
+    kind: Issuer
+    {{- end }}
+  secretName: yugabyte-{{ .name }}-tls
```

Comment:
For now, we can match the current behavior of generating separate certificates for masters and tservers. This will make it easier to change in the future (all at once) if necessary.